--- a/config.def.h	2024-09-28 14:46:27.227010221 +0200
+++ b/config.def.h	2024-09-28 14:45:03.323341010 +0200
@@ -1,6 +1,6 @@
 /* user and group to drop privileges to */
 static const char *user  = "nobody";
-static const char *group = "nogroup";
+static const char *group = "nobody";
 
 static const char *colorname[NUMCOLS] = {
 	[INIT] =   "black",     /* after initialization */
@@ -11,5 +11,8 @@
 /* treat a cleared input like a wrong password (color) */
 static const int failonclear = 1;
 
+/* lock screen opacity */
+static const float alpha = 0.3;
+
 /* time in seconds to cancel lock with mouse movement */
 static const int timetocancel = 4;
--- a/slock.c	2024-09-28 14:46:19.066977566 +0200
++ b/slock.c	2024-09-28 14:44:54.923307367 +0200
@@ -23,4 +23,5 @@
#include <X11/Xutil.h>
#include <X11/Xft/Xft.h>
+#include <X11/Xatom.h>
#include <X11/XF86keysym.h>

@@ -294,7 +294,10 @@
 				XRRSelectInput(dpy, lock->win, RRScreenChangeNotifyMask);
 
 			XSelectInput(dpy, lock->root, SubstructureNotifyMask);
-			locktime = time(NULL);
+			locktime = time(NULL); 
+			unsigned int opacity = (unsigned int)(alpha * 0xffffffff);
+			XChangeProperty(dpy, lock->win, XInternAtom(dpy, "_NET_WM_WINDOW_OPACITY", False), XA_CARDINAL, 32, PropModeReplace, (unsigned char *)&opacity, 1L);
+		        XSync(dpy, False);
 			return lock;
 		}
 
